front/src/common/Map/Search/MapSearchStation.tsx

```diff
@@ -43,17 +48,34 @@ const MapSearchStation = ({ updateExtViewport, closeMapSearchPopUp }: MapSearchS
 
   // Create playload based on the type of search "name" or "trigram"
 
-  const createPayload = (searchQuery: SearchQuery) => ({
-    object: 'operationalpoint',
-    query: ['and', searchQuery, infraID !== undefined ? ['=', ['infra_id'], infraID] : true],
-  });
+  const createPayload = (searchQueryList: SearchQuery[]) => {
+    const searchQuery =
+      searchQueryList.length === 1 ? searchQueryList[0] : ['and', ...searchQueryList];
+    return {
+      object: 'operationalpoint',
+      query: ['and', searchQuery, infraID !== undefined ? ['=', ['infra_id'], infraID] : true],
+    };
+  };
+
+  const createQuery = (isSearchByTrigram?: boolean) =>
+    Object.keys(searchState).reduce((acc, key) => {
+      const searchKey = key as keyof SearchState;
+      if (searchState[searchKey]) {
+        if (key === 'name' && isSearchByTrigram)
+          return [...acc, ['=i', ['trigram'], searchState.name]];
```

Comment:
don’t we already have a function that create trigrams like this ?

Reply:
I didn't find such function, so i added one for the `MapSearchStation`, `MapSearchLine` and `MapSearchSignal` components as it was relatively the same code :)